scenedetect/video_manager.py

```diff
@@ -743,8 +743,13 @@ def read(self):
         if self._curr_cap is not None and self._end_of_video != True:
             while not read_frame:
                 read_frame, self._last_frame = self._curr_cap.read()
-                if not read_frame and not self._get_next_cap():
-                    break
+                # Switch to the next capture when the current one is over
+                if not read_frame:
```

Comment:
I think this should be changed to `while not read_frame` instead now, since if for some reason the new cap doesn't return a frame, it will still be used below.  Changing this to a while() would advance to the next cap, if any.

Reply:
Sorry I realized just now, a while loop is not the solution actually - have a better suggestion.